File: Known.Web/src/Frame/Core/sys/SysNotice.js
function SysNotice() {
    var type = Utils.getUrlParam('type'),
        url = {
            QueryModels: baseUrl + '/System/QueryNotices',
            DeleteModels: baseUrl + '/System/DeleteNotices',
            SaveModel: baseUrl + '/System/SaveNotice',
            PublishNotice: baseUrl + '/System/PublishNotice'
        };

    var view = new View('SysNotice', {
        url: url,
        columns: [
            { field: 'Id', type: 'hidden' },
            {
                title: Language.Title, field: 'Title', query: true, format: function (d) {
                    return $('<span class="link">')
                        .append(d.Title)
                        .data('data', d)
                        .click(function () {
                            var data = $(this).data('data');
                            Layer.open({
                                title: data.Title, width: 960, height: 500,
                                content: data.Content
                            });
                        });
                },
                type: 'text', required: true, lineBlock: true, inputStyle: 'width:50%'
            },
            { onlyForm: true, title: Language.Content, field: 'Content', type: 'editor', required: true, lineBlock: true },
            { title: Language.Status, field: 'Status' },
            { title: Language.PublishBy, field: 'PublishBy', type: 'text' },
            { title: Language.PublishTime, field: 'PublishTime', placeholder: 'yyyy-MM-dd HH:mm', type: 'date' }
        ],
        formToolbar: [{
            text: Language.Publish, icon: 'fa fa-send-o', handler: function (e) {
                e.form.save(url.PublishNotice, function () {
                    view.loadGrid();
                });
            }
        }],
        gridOption: {
            where: { type: type }
        }
    });

    //methods
    this.render = function (dom) {
        view.render().appendTo(dom);
    }

    this.mounted = function () {
        view.load();
    }
}

$.extend(Page, {
    SysNotice: { component: new SysNotice() }
});

File: Known/Entities/SysNotice.cs
using System;

namespace Known.Entities
{
    /// <summary>
    /// 通知公告实体类。
    /// </summary>
    public class SysNotice : EntityBase
    {
        /// <summary>
        /// 取得或设置状态（暂存，已发布）。
        /// </summary>
        [Column("状态", "", true, "1", "50")]
        public string Status { get; set; }

        /// <summary>
        /// 取得或设置标题。
        /// </summary>
        [Column("标题", "", true, "1", "50")]
        public string Title { get; set; }

        /// <summary>
        /// 取得或设置内容。
        /// </summary>
        [Column("内容", "", false, "1", "4000")]
        public string Content { get; set; }

        /// <summary>
        /// 取得或设置发布人。
        /// </summary>
        [Column("发布人", "", false, "1", "50")]
        public string PublishBy { get; set; }

        /// <summary>
        /// 取得或设置发布时间。
        /// </summary>
        [Column("发布时间", "", false)]
        public DateTime? PublishTime { get; set; }
    }
}